executor/src/witgen/generator.rs

```diff
@@ -173,6 +203,10 @@ where
 
         self.shift_rows();
 
+        if next_row == 0 {
+            self.first = self.previous.clone()
```

Comment:
Don't really understand why this. "If the next row is the first, copy the second last"?

Reply:
It's confusing because this is after `self.shift_rows`

Reply:
Also `next_row` is actually `current_row`

Reply:
Yeah, I think the `next_row` should be renamed... (Fun fact: It's the same as `self.current_row_index`)

But for this PR, we could just move this block before `self.shift_row()`, right?